### PR TITLE
added shell sort

### PR DESCRIPTION
**Pull request by sn99**
#3 Added shell sort with tests

### CODE REVIEW

**Comment by efugier** [comment]
Nice, please run `cargo fmt` before commiting though.

**Comment by sn99** [comment]
@efugier It's done

**Comment by efugier** [comment]
One more step and you're done: the policy has changed, please write your tests in `shell_sort.rs` !

**Comment by sn99** [comment]
@efugier it's done

**Comment by AnshulMalik** [comment]
This is great! @sn99  Just a couple of suggestions, 

- Some comments will help a lot
- `sort_gap_insertation` never heard of insertation :P, did you mean insertion?
- A rebase :)

**Comment by sn99** [comment]
@AnshulMalik @efugier it's done

File: src/sorting/shell_sort.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
+pub fn shell_sort<T: Ord + Copy>(values: &mut Vec<T>) {
+    // shell sort works by swiping the value at a given gap and decreasing the gap to 1
```
Comment (pickfire) [inline src/sorting/shell_sort.rs]: Do you mean `swapping`?

File: src/sorting/shell_sort.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    fn insertion<T: Ord + Copy>(values: &mut Vec<T>, start: usize, gap: usize) {
+        for i in ((start + gap)..values.len()).step_by(gap) {
+            let val_current = values[i];
+            let mut pos = i;
+            // make swaps
+            while pos >= gap && values[pos - gap] > val_current {
+                values[pos] = values[pos - gap];
```
Comment (pickfire) [inline src/sorting/shell_sort.rs]: Maybe you might want to take a look [`slice.swap()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.swap)?

**Comment by AnshulMalik** [comment]
Nice catch @pickfire, Maybe we should fix them :)

**Pull request merged**